### PR TITLE
Parsec PsaGenerateRandom operation implementation for CryptoAuthLib provider

### PR DESCRIPTION
**Pull request by RobertDrazkowskiGL**
An implementation of a psa_generate_random callback inside a cryptoauthlibrary-provider responsible for PsaGenerateRandom operation/opcode using CryptoAuthentication Library API calls.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -46,7 +46,6 @@ libc = "0.2.77"
 anyhow = "1.0.32"
 # Using a fork until the JWT support is merged into the main rust-spiffe repository
 spiffe = { git = "https://github.com/hug-dev/rust-spiffe", branch = "refactor-jwt" }
-#rust-cryptoauthlib = { git = "https://github.com/RobertDrazkowskiGL/rust-cryptoauthlib.git", optional = true }
```
Comment (hug-dev) [inline Cargo.toml:49]: 🧹

File: src/providers/cryptoauthlib/generate_random.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                let mut buffer = Vec::with_capacity(rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE);
+                let err = rust_cryptoauthlib::atcab_random(&mut buffer);
+                match err {
+                    rust_cryptoauthlib::AtcaStatus::AtcaSuccess => {
+                        // append buffer vector to result vector
+                        random_bytes.append(&mut buffer);
+                    }
+                    _ => {
+                        let error = ResponseStatus::PsaErrorGenericError;
+                        format_error!("Hash computation failed ", err);
```
Comment (hug-dev) [inline src/providers/cryptoauthlib/generate_random.rs:34]: Probably copied code from hash here, `"Hash computation failed"`

**Review by hug-dev — COMMENTED** [review]
Looks good! Just a small thing I think could improve the algorithm

File: src/providers/cryptoauthlib/generate_random.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        // size 0 - no need to run operation
+        if op.size == 0 {
+            Ok(psa_generate_random::Result {
+                random_bytes: Zeroizing::new(vec![]),
+            })
+        }
+        // for size bigger than buffer size operation need to be executed multiple times
+        else {
+            // calculate loop count
+            let call_count = (op.size / rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE) + 1;
```
Comment (hug-dev) [inline src/providers/cryptoauthlib/generate_random.rs:22]: Forgot to put it in the review 🤦 

Sorry to nerd out, but I think you can remove the `if`/`else` if you calculate `call_count` as:
```rust
let call_count = (op.size + rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE - 1) / rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE;
```

Also a bit more efficient for when `op.size` is a multiple of `rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE` 🧠

Comment (ionut-arm) [reply src/providers/cryptoauthlib/generate_random.rs:22]: ? if `op.size` is equal to `rust_cryptoauthlib::ACTA_RANDOM_BUFFER_SIZE`, then your formula sets `call_count` to 1, whereas the existing one sets it to 2

**Review by hug-dev — APPROVED** [review]
Thanks! Looks good to me.

**Review by ionut-arm — APPROVED** [review]

**Comment by hug-dev** [comment]
Spurious errors with PKCS11, I am going to push another PR soon and will fix any errors there if they remain.

**Pull request merged**